pkg/build/metrics/prometheus/metrics_test.go

```diff
@@ -0,0 +1,138 @@
+package prometheus
+
+import (
+	"bytes"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	buildapi "github.com/openshift/origin/pkg/build/apis/build"
+	internalversion "github.com/openshift/origin/pkg/build/generated/listers/build/internalversion"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type fakeLister []*buildapi.Build
+
+func (f fakeLister) List(labels.Selector) ([]*buildapi.Build, error) {
```

Comment:
why building something like this when you can use the "real" fake lister ;-)

see: https://github.com/openshift/origin/pull/16657/commits/1418d744a0a1dcb41b8fd730106dceddd643ccd1#diff-94b6ca7b6dc08be688abf41c64447a29R198 for example

Reply:
is `"real" fake` the epitome of an oxymoron :-) ?

I was unaware that you could add to the cache in this manner.  @jim-minter - since my unit test here is a shameless derivative of your TSB metrics unit test, I'll pull you in - were you aware of the pattern @mfojtik highlights?  And if so, did you have a reason for not employing it?

At first blush, not being the "real" fake aside, the current approach seems simpler in setup, but what @mfojtik 's used does not involve the use of the prometheus http support, so probably has less path length to get to the apps code he wants to test.

Color me on the fence.

And would @bparees rather merge what we have here now and debate this impl detail later on?

Reply:
@bparees would definitely prefer to merge what we have here now.

and yeah, this approach seems simpler.